DBZ-8318 Explain how to manage SPI and bundled jars for debezium engine applications (#5933)
debezium/debezium: documentation/modules/ROOT/pages/development/engine.adoc

The remainder of this document describes embedding the MySQL connector in your application.
Other connectors are used in a similar manner, except with connector-specific configuration, topics, and events.

=== Package you project
Debezium uses the SPI via the ServiceLoader to load different implementations depending on the connector type or for custom implementations.

When an interface has multiple implementations — for instance, `io.debezium.snapshot.spi.SnapshotLock`, which has a default implementation in the core and specific ones for each connector — you must explicitly configure your build tools to merge the `META-INF/services files`.
If this step is missed, Debezium may fail to locate the required implementation.

> To address this, instruct your build tools accordingly. For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
> you need to add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer:

Reviewer: ```suggestion
For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer, as shown in the following example:
```

[source,xml]
----
...
<configuration>
 <transformers>
    ...
    <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
    ...
 </transformers>
...
</configuration>
----

> Alternatively, if using the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in] you can utilize the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].

Reviewer: ```suggestion
Alternatively, if you use the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in], you can use the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```
